indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp

```diff
@@ -0,0 +1,1799 @@
+    //
+    optional<float> GetAnalogueGain() const {
+        if (!m_cameraAcquired) {
+            INDIGO_DRIVER_ERROR(DRIVER_NAME, "Camera not acquired" );
+            return {};
+        }
+        return m_controls.get(controls::AnalogueGain);
+    }
+    //
+    optional<float> GetDigitalGain() const {
+        if (!m_cameraAcquired) {
+            INDIGO_DRIVER_ERROR(DRIVER_NAME, "Camera not acquired" );
+            return nullopt;
+        }
+        return m_controls.get(controls::DigitalGain);
+    }
+    bool IsRequestCreated() const {
+        return !m_requests.empty();
+    } 
+    bool IsAnyRequestComplete() const {
+        return std::any_of(m_requests.begin(), m_requests.end(),
+            [](const Request_p &req) { return req->status() == Request::RequestComplete; });
+    }
+    Size GetMaxImageSize() const;
+    FrameInfo GetFrameInfo(const PixelFormat &format);
+    optional<float> GetSensorTemperature() const { return m_controls.get(controls::SensorTemperature); }
+    // Controls
+    //
+    int  ConfigureCamera();
+    
+    // Operations
+    //
+    int StartExposure(int width, int height, PixelFormat format);
+    int StartCamera();
+    void ReturnFrameBuffer(FrameData frameData);
+    void Cleanup() {
+        if (m_camera) {
+            m_camera->stop();
+            m_camera->release();
+            m_camera.reset();          
+        }
+        m_requests.clear();
+        m_config.reset();
+
+    }
+    //
+    void Set(ControlList controls);
+    void StopCamera();
+    void CloseCamera();
+    void ProcessRequest(Request *request);
+    // Camera auto-focus trigger 
+    //
+    bool af();
+    //    
+    ~RPiCamera(){}
+   private:
+    // Camera states
+    //
+    bool m_cameraAcquired = false;
+    bool m_cameraStarted = false;
+    
+    Camera_p m_camera{};
+    Config_p m_config{};
+    Requests_p m_requests{};
+    std::unique_ptr<FrameBufferAllocator> m_allocator;
+
+    Stream *m_stream = nullptr;
+    std::string cameraId;
+    PixelFormats m_pixelFormats{libcamera::formats::SRGGB8}; // Default pixel format
+    //
+    bool m_afEnabled = false;
+    //
+    std::queue<Request *> m_requestQueue;
+    std::map<int, std::pair<void *, unsigned int>> m_mappedBuffers;
+    ControlList m_controls;
+    std::mutex m_controlMutex;
+    std::mutex m_cameraStopMutex;
+    std::mutex m_freeRequestsMutex;
+    //
+    void ConfigureStill(int width, int height, PixelFormat format);
+    int StartCapture();
+    int QueueRequest(Request *request);  
+    void StreamDimensions(Stream const *stream, uint32_t *w, uint32_t *h, uint32_t *stride) const;
+    void Reset();
+ };
+
+ bool RPiCamera::IsMonochrome() const{
+    if (!m_cameraAcquired) {
+        INDIGO_DRIVER_ERROR(DRIVER_NAME, "Camera not acquired" );
+        return false;
+    }
+
+    return properties::draft::ColorFilterArrangementEnum::MONO == 
+        m_camera->properties().get(properties::draft::ColorFilterArrangement); 
+ }
```

Comment:
Please fix indentations!

You mix 2 spaces with 4 tabs as indentations - use tabs only!